### DISCUSSION TITLE
Find combobox item in visual tree as Snoop do it

### DISCUSSION THREAD

**Discussion by lsoft** [snoopwpf/snoopwpf#389] [Q&A] [answered]
Hi! 

I need to find combobox item in the visual tree inside my app. Everything is local, no IPC. My target is to build a visual tree like Snoop do it, but I have difficulties to process combobox items. Even I open th e combobox, there are no items in the visual tree.

I workaround it with `ItemContainerGenerator.ContainerFromItem`, looks like it works. But I'm not sure it's a correct way.

I wonder is the tree builder code open source? I failed to find it. If not, I would want to ask for any tips, how to build the similar tree in my code.

Thanks for this project! It helped me a lot!

**Comment by batzen** [comment, marked as answer]
Sorry for my late response.

The items are not immediate children of the visual tree of the ComboBox, but of it's popup. That's why you can't find them there.

Everything of Snoop is OSS and on GitHub. Nothing is hidden or added only during the CI build.
The code you are looking for is at https://github.com/snoopwpf/snoopwpf/blob/262f0b07e9478e3f4e3266db6bb5335993381cb8/Snoop.Core/Data/Tree/TreeService.cs#L143

The only thing that's kind of hidden is the digital signing of the artifacts, because there is other way to digitally sign.

**Reply by lsoft** [reply]
@batzen thank you!
